go/worker/client/committee/node.go

```diff
@@ -366,14 +378,15 @@ func (n *Node) worker() {
 }
 
 // NewNode creates a new client node.
-func NewNode(commonNode *committee.Node) (*Node, error) {
+func NewNode(commonNode *committee.Node, roleProvider registration.RoleProvider) (*Node, error) {
```

Comment:
```suggestion
func NewNode(commonNode *committee.Node, opts ...NodeOption) (*Node, error) {
```
Optional parameters should be specified differently, e.g. via variadic options. It is "ugly" to accept `nil` parameters.

Reply:
I disagree :P  This is just a single option, it makes no sense to add another layer of abstraction just to pass that.  If we have more optional flags for `NewNode` in the future, we can add the option struct then.

Reply:
I disagree again :D Once you allow accepting `nil` parameters, even if that is only one, programmers should start questioning themselves which fields can be `nil` and which not. In general, that should be forbidden, but if you allow it, it should be explicit (e.g. with options).

Reply:
:D
I mean, this part of the code is not public-facing and the executor node does it the same way :P Adding the role provider as an option would just overcomplicate the rest of the code with needless boilerplate -- `NewNode` is only called from within this file.  In this particular case, I think it's OK and not confusing.

Reply:
This kind of questions arise if you allow nil values and have no idea if some fields can be `nil`:
https://github.com/oasisprotocol/oasis-core/pull/6050/files#r1950633603